packages/type-jsonrpc/src/author.ts

```diff
@@ -22,8 +22,22 @@ const submitExtrinsic: MethodOpt = {
   type: 'Hash'
 };
 
+const extrinsicUpdate: MethodOpt = {
+  description: 'Subscribe and watch an extrinsic until unsubscribed',
+  isSigned: true,
+  isSubscription: true,
+  subscribe: [
+    'author_submitAndWatchExtrinsic',
+    'author_unwatchExtrinsic'
+  ],
+  params: [
+    createParam('Bytes')
+  ],
+  type: 'ExtrinsicUpdate'
```

Comment:
Now it gets tricky -

1. Match the type here to the Rust code, it should be `Status` (since it is RPC, we could name it slightly more descriptive, e.g `ExtrinsicStatus` - the first version . is a bit "generic") - https://github.com/paritytech/substrate/blob/master/core/rpc/src/author/mod.rs#L67 (without the generic params)
2. Implement the actual type class inside `api-codec` - since it is a typed enum, it is a bit more tricky. Basically you need `extends EnumType<class | class | class>` where `class` is the actual classes. (These can be classes inside the `ExtrinsicStatus` file, e.g. see https://github.com/polkadot-js/api/blob/master/packages/api-codec/src/MisbehaviorReport.ts#L57)

You had to pick a difficult one ;) (Metadata also has an EnumType, that one is probably be a bit cleaner since it doesn't have the enum overrides, i.e. it starts at index 0, is the above report it starts at 0x11 which just makes things a bit more complicated)